core/src/components/badge/badge.ionic.scss

```diff
@@ -175,18 +182,33 @@
   height: globals.$ion-scale-400;
 }
 
+// Badge Sizes Not Empty With Vertical Prop
+// --------------------------------------------------
+
+:host(.badge-small[vertical]:not(:empty)),
+:host(.badge-medium[vertical]:not(:empty)),
+:host(.badge-large[vertical]:not(:empty)) {
+  @include globals.typography(globals.$ion-body-sm-medium);
+  min-width: globals.$ion-scale-400;
+  height: globals.$ion-scale-400;
+}
```

Comment:
We shouldn't be changing the size of the badges. Based on the design, the `xxsmall` should be used.

```suggestion
```

Reply:
Fixed